### PR TITLE
Enable AITL for latest Azure images

### PR DESCRIPTION
**Pull request by linguini-dev**
This PR enables AITL (Azure Image Testing for Linux). This test module creates AITL Jobs in Azure. 
- Related ticket: https://progress.opensuse.org/issues/168334
- Additional Info: https://github.com/microsoft/lisa
- Verification run: http://dirtman.qe.prg2.suse.org/tests/235

### CODE REVIEW

File: tests/publiccloud/azure_aitl.pm
Comment (ricardobranco777) [inline tests/publiccloud/azure_aitl.pm:3]:
```suggestion
# Copyright 2024 SUSE LLC
```

File: tests/publiccloud/azure_aitl.pm
Comment (ricardobranco777) [inline tests/publiccloud/azure_aitl.pm:67]: I think you want to use record_info here.  Otherwise there's no reason to call script_output here.

File: data/publiccloud/aitl/shared_gallery.json
Comment (asmorodskyi) [reply data/publiccloud/aitl/shared_gallery.json:9]: this should not block merge just question for future improvements ...

File: lib/main_publiccloud.pm
Comment (asmorodskyi) [inline lib/main_publiccloud.pm:165]: this line needs to be removed , I assume you simply forgot to tweak this part ...

File: tests/publiccloud/azure_aitl.pm
Comment (asmorodskyi) [inline tests/publiccloud/azure_aitl.pm:29]: we need to discuss this part , we should rely on the fact that tools image provide everything what you need

**Review by asmorodskyi — CHANGES_REQUESTED** [review]
generally all looks good but there are two major issues which we need to solve before it can be merged : 
1. triggering logic needs update (main_publiccloud)
2. logic about az installation have several problems ( let's discuss it in person )

File: tests/publiccloud/azure_aitl.pm
Comment (asmorodskyi) [inline tests/publiccloud/azure_aitl.pm:39]:
```suggestion
    my $region = get_var('PUBLIC_CLOUD_REGION');
```

this is design decision done on purpose . It is responsibility of triggering logic / job group settings to define in which region test will run . And if region is not defined test MUST fail

File: tests/publiccloud/azure_aitl.pm
Comment (linguini-dev) [reply tests/publiccloud/azure_aitl.pm:67]: Actually I think I added this just as a debugging step and it can be removed altogether.

File: data/publiccloud/aitl/shared_gallery.json
Comment (linguini-dev) [reply data/publiccloud/aitl/shared_gallery.json:9]: There is not option to run whatever (at the time of writing). A test manifest must be specified and it has some parameters. They have added 2 new tests since I wrote this, so your gut feeling is correct :)  

As a side note - we want to keep track if the tests pass or not, and adding new tests without our knowledge (because MS updated something) can be annoying to debug.

File: lib/main_publiccloud.pm
Comment (linguini-dev) [reply lib/main_publiccloud.pm:165]: Indeed. The whole test needs to  be moved elsewhere, I forgot to move it.

File: tests/publiccloud/azure_aitl.pm
Comment (linguini-dev) [reply tests/publiccloud/azure_aitl.pm:29]: I just copied from the `azure_cli` test module, so let's discuss.

File: tests/publiccloud/azure_aitl.pm
Comment (linguini-dev) [reply tests/publiccloud/azure_aitl.pm:39]: Sure, but in the specific instance for AITL, it kind of doesn't matter. 

I couldn't get AITL jobs running anywhere but US (westus1,2,3 and etc.) so for the time being, the region is hardcoded.

File: data/publiccloud/aitl/shared_gallery.json
Comment (linguini-dev) [reply data/publiccloud/aitl/shared_gallery.json:9]: It might make sense to import the tests based on Tiers instead of by caseName. Looking into that.

File: tests/publiccloud/azure_aitl.pm
Comment (linguini-dev) [reply tests/publiccloud/azure_aitl.pm:39]: We also do this: https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/7dab163d5ce61f9cbd3047677c3ad29c92eab956/lib/publiccloud/k8sbasetest.pm#L91 So if nothing is specified it will be set to `westeurope` anyway :shrug:

**Comment by linguini-dev** [comment]
I caught some issues with the While loop not triggering and not generatin the Failed xml correctly

File: tests/publiccloud/azure_aitl.pm
Comment (linguini-dev) [reply tests/publiccloud/azure_aitl.pm]: Removed this part as it was unnecessary.

File: tests/publiccloud/azure_aitl.pm
Comment (asmorodskyi) [inline tests/publiccloud/azure_aitl.pm:51]: can you elaborate how this line correlate with this section in scenaion JSON ? 
```
           "region": [
                "westus",
                "westus2",
                "westus3"
            ],
```

so which region(s) will be used after all ?

File: tests/publiccloud/azure_aitl.pm
Comment (asmorodskyi) [reply tests/publiccloud/azure_aitl.pm]: so why you need this variable at all ? just drop it

File: tests/publiccloud/azure_aitl.pm
Comment (asmorodskyi) [inline tests/publiccloud/azure_aitl.pm:78]: maybe we need to replace this with `record_info` ?

File: tests/publiccloud/azure_aitl.pm
Comment (asmorodskyi) [inline tests/publiccloud/azure_aitl.pm:96]: Hi @mdoucha . AFAIK you was involved in writing parser for LTP tests . Can you please review this part from POV "Extra results tab parsers" ? 

CC: @foursixnine , @perlpunk

File: tests/publiccloud/azure_aitl.pm
Comment (linguini-dev) [reply tests/publiccloud/azure_aitl.pm]: Not needed. This helps when you check the Live Log on the job, and you can see how many Jobs are scheduled and how much you have to wait for. If it was Record Info, it would appear in the end, at which point you would already know this.

File: tests/publiccloud/azure_aitl.pm
Comment (linguini-dev) [reply tests/publiccloud/azure_aitl.pm:51]: TL;DR; Curently the AITL Job Template resource or whatever its called in Azure is available in the WestUS region only (from what I've seen). 

So the AITL Jobs have to be scheduled in that region, however from testing locally I have seen that the resource groups and other objects can be created wherever.

File: tests/publiccloud/azure_aitl.pm
Comment (rbmarliere) [reply tests/publiccloud/azure_aitl.pm:34]: Yes, but what happens when that version is obsolete? Does it even need to be frozen like that, is using "master" too risky?

Comment (linguini-dev) [reply tests/publiccloud/azure_aitl.pm:34]: MS are notorious for changing things rapidly and AITL / LISA seems to be very BETA / experimental at this stag.

 This version is actually tagged in the repo: https://github.com/microsoft/lisa/releases so I doubt it will disappear before we update it. e.g. The oldest available tag in the drop-down in GIthub is from `2022` so I think we're good.

File: tests/publiccloud/azure_aitl.pm
Comment (asmorodskyi) [inline tests/publiccloud/azure_aitl.pm:52]:
```suggestion
    assert_script_run("az group create -n $resource_group -l $region  --tags '$tags'");
```

and you can drop line above

File: data/publiccloud/aitl/shared_gallery.json
Comment (linguini-dev) [reply data/publiccloud/aitl/shared_gallery.json]: After discussing with our contact, it does in fact make sense to use Tiers https://mslisa.readthedocs.io/en/main/run_test/microsoft_tests.html#test-tier instead of specific caseNames. However, our initial sample cases were ~40 tests while T2 for example are 180+. So I would like to try using T1 for the time being and we can later either cherry-pick or increase/decrease the Tier.

**Review by asmorodskyi — APPROVED** [review]
LGTM

**Pull request merged**
